diff --git a/subsys/usb/device/class/msc.c b/subsys/usb/device/class/msc.c
index 1971c942af8..3e5978ecb6b 100644
--- a/subsys/usb/device/class/msc.c
+++ b/subsys/usb/device/class/msc.c
@@ -423,6 +423,13 @@ static bool modeSense6(void)
 	return write(sense6, sizeof(sense6));
 }
 
+static bool modeSense10(void)
+{
+	uint8_t sense10[] = { 0x00, 0x06, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+
+	return write(sense10, sizeof(sense10));
+}
+
 static bool readFormatCapacity(void)
 {
 	uint8_t capacity[] = { 0x00, 0x00, 0x00, 0x08,
@@ -602,6 +609,12 @@ static void CBWDecode(uint8_t *buf, uint16_t size)
 				modeSense6();
 			}
 			break;
+		case MODE_SENSE10:
+			LOG_DBG(">> MODE_SENSE10");
+			if (check_cbw_data_length()) {
+				modeSense10();
+			}
+			break;
 		case READ_FORMAT_CAPACITIES:
 			LOG_DBG(">> READ_FORMAT_CAPACITY");
 			if (check_cbw_data_length()) {
